src/main/java/jenkins/plugins/http_request/HttpRequest.java

```diff
@@ -50,54 +54,88 @@
+    private String outputFile                 = DescriptorImpl.defaultOutputFile;
+    private Integer timeout                   = DescriptorImpl.defaultTimeout;
+    private Boolean consoleLogResponseBody    = DescriptorImpl.defaultConsoleLogResponseBody;
+    private String authentication             = DescriptorImpl.defaultAuthentication;
+    private List<NameValuePair> customHeaders = DescriptorImpl.defaultCustomHeaders;
+
+    private TaskListener listener;
 
     @DataBoundConstructor
-    public HttpRequest(String url, HttpMode httpMode, String authentication, MimeType contentType,
-                       MimeType acceptType, String outputFile, Boolean returnCodeBuildRelevant,
-                       Boolean consoleLogResponseBody, Boolean passBuildParameters,
-                       List<NameValuePair> customHeaders, Integer timeout,
-                       String validResponseCodes, String validResponseContent)
-            throws URISyntaxException {
+    public HttpRequest(@Nonnull String url) {
         this.url = url;
-        this.contentType = contentType;
-        this.acceptType = acceptType;
-        this.outputFile = outputFile;
+    }
+
+    @DataBoundSetter
+    public void setHttpMode(HttpMode httpMode) {
         this.httpMode = httpMode;
-        this.customHeaders = customHeaders;
+    }
+
+    @DataBoundSetter
+    public void setPassBuildParameters(Boolean passBuildParameters) {
+        this.passBuildParameters = passBuildParameters;
+    }
+
+    @DataBoundSetter
+    public void setValidResponseCodes(String validResponseCodes) {
         this.validResponseCodes = validResponseCodes;
+    }
+
+    @DataBoundSetter
+    public void setValidResponseContent(String validResponseContent) {
         this.validResponseContent = validResponseContent;
-        this.authentication = Util.fixEmpty(authentication);
-        this.consoleLogResponseBody = consoleLogResponseBody;
-        this.passBuildParameters = passBuildParameters;
+    }
+
+    @DataBoundSetter
+    public void setAcceptType(MimeType acceptType) {
+        this.acceptType = acceptType;
+    }
+
+    @DataBoundSetter
+    public void setContentType(MimeType contentType) {
+        this.contentType = contentType;
+    }
+
+    @DataBoundSetter
+    public void setOutputFile(String outputFile) {
+        this.outputFile = outputFile;
+    }
+
+    @DataBoundSetter
+    public void setTimeout(Integer timeout) {
         this.timeout = timeout;
+    }
 
-        this.returnCodeBuildRelevant = returnCodeBuildRelevant;
+    @DataBoundSetter
+    public void setConsoleLogResponseBody(Boolean consoleLogResponseBody) {
+        this.consoleLogResponseBody = consoleLogResponseBody;
+    }
+
+    @DataBoundSetter
+    public void setAuthentication(String authentication) {
+        this.authentication = authentication;
+    }
+
+    @DataBoundSetter
+    public void setCustomHeaders(List<NameValuePair> customHeaders) {
```

Comment:
(For proper databinding, instead of the hand-coded `<f:repeatable field="customHeaders">` block in `HttpRequest/config.jelly`, you would have a simple `f:repeatableProperty`.)